<html lang="en">
<head>
    <title>GenePattern Resources</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">

    <!-- Style -->
    <link type="text/css" rel="stylesheet" href="style.css">
</head>
<body>
    <div class="main-container">
        <div class="header">
            <h1>GenePattern Resources</h1>
            <hr>
        </div>
        <div class="toc">
            <h4>Table of Contents</h4>
            <ol>
                <li><a href="#get-started">Get Started</a></li>
                <li><a href="#importing-data">Available Tools</a></li>
                <ol>
                    <li><a href="#importing-data">Importing Data</a></li>
                    <li><a href="#gene-expression">Gene Expression Analysis</a></li>
                    <li><a href="#single-cell">Single-cell Sequencing</a></li>
                    <li><a href="#gsea">GSEA</a></li>
                    <li><a href="#mutation">Mutational Significance Analysis</a></li>
                    <li><a href="#proteomics">Proteomics</a></li>
                    <li><a href="#pathway">Pathway Analysis</a></li>
                    <li><a href="#epigenetics">Epigenetics</a></li>
                    <li><a href="#snp">SNP and Copy Number Analysis</a></li>
                    <li><a href="#visual">Visualization Tools</a></li>
                </ol>
                <li><a href="">Contributors</a></li>
            </ol>
            <hr>
        </div>
        <div class="get-started" id="get-started">
            <h2>Get Started</h2>
            <br>
            <h4>What is GenePattern?</h4>
            <p>GenePattern is a freely available open-source scientific workflow system that is designed for computational biology. It provides access to a repository of hundreds of bioinformatics analysis and visualization methods (“modules”), as well as utilities for data formatting, preprocessing, and other auxiliary functions that provide important glue between analysis steps. The user interface is point and click with no programming required. The public GenePattern server, hosted since 2008, has tens of thousands of registered users and runs thousands of analysis jobs per week. The software has been downloaded for local installation by bioinformatics core facilities, research laboratories, and individual scientists worldwide.

            </p>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tutorial</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="https://www.genepattern.org/quick-start">GenePattern Quickstart</a></th>
                        <td>This brief tutorial introduces you to GenePattern by providing step-by-step instructions for running an analysis and viewing the results. In less than 10 minutes you'll run your first analysis and review the results.</td>
                    </tr>
                    <tr>
                        <th><a href="https://www.genepattern.org/tutorial">Differential Expression Tutorial</a></th>
                        <td>The GenePattern Tutorial introduces you to GenePattern by providing step-by-step instructions for analyzing gene expression. It takes approximately 40 minutes to complete.
                        </td>
                    </tr>
                    <tr>
                        <th><a href="https://www.genepattern.org/concepts">Concepts</a></th>
                        <td>This Concepts guide provides a brief introduction to GenePattern. All other GenePattern documentation assumes that you are familiar with the concepts covered here.</td>
                    </tr>
                </tbody>
            </table>


            


            <h4>What is GenePattern Notebook</h4>
            <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Fuga quod aut asperiores suscipit? Molestias dolorem officia voluptas ex atque dolor! Inventore, atque tenetur. Neque quo fugiat nisi hic eos provident.</p>

            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tutorial</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="https://notebooks.genepattern.org/services/sharing/notebooks/361/preview/">GenePattern Notebook Tutorial</a></th>
                        <td>A tutorial on how to use the basic features of the GenePattern Notebook environment.
                        </td>
                    </tr>
                    <tr>
                        <th><a href="https://notebooks.genepattern.org/services/sharing/notebooks/294/preview/">UI Builder Tutorial</a></th>
                        <td>A tutorial of how to use the UI Builder, aimed at notebook authors with programmatic skill.</td>
                    </tr>
                    <tr>
                        <th><a href="https://notebooks.genepattern.org/services/sharing/notebooks/27/preview/">Run an Analysis</a></th>
                        <td>A simple example of how to run an analysis in GenePattern Notebook, in which you preprocess gene expression data and use a heat map to visualize the results.</td>
                    </tr>
                    <tr>
                        <th><a href="https://notebooks.genepattern.org/services/sharing/notebooks/19/preview/">GenePattern Python Tutorial</a></th>
                        <td>A tutorial on how to programmatically call GenePattern services from Python.</td>
                    </tr>
                </tbody>
            </table>
            <!-- <div style="border: 1px solid grey; padding: 20px; margin:50px; overflow-y: scroll; background: url('banner2.jpg'); text-align:left; margin-top: 0px;margin-bottom: 0px;">
                <h1 style="margin-bottom: 20px; margin-top: 0px; font-size: 30px; color: white;">Tutorial Notebooks</h1>
                <div id="notebook_list" class="scrollable-menu" role="menu" style="max-height: 400px; overflow-x: hidden; overflow-y: scroll; margin-right: 1px;">
                </div>
                <script type="application/javascript">
                    function workshop(nb) {
                        let is_workshop = false;
                        nb.tags.forEach(function(tag) {
                            if (tag.label === "workshop") is_workshop = true;
                        });

                        return is_workshop;
                    }

                    function tutorial(nb) {
                        let is_tutorial = false;
                        nb.tags.forEach(function(tag) {
                            
                            if (tag.label === "tutorial") is_tutorial = true;
                        });

                        return is_tutorial;
                    }

                    function sort_notebooks(notebooks) {
                        const tutorial1 = [];
                        const tutorial_nb = [];

                        notebooks.forEach(function(nb) {
                            if (nb.name === "GenePattern Notebook Tutorial") tutorial1.push(nb);
                            else if (tutorial(nb)) tutorial_nb.push(nb);
                        });

                        return tutorial1.concat(tutorial_nb);
                    }

                    $.ajax({
                        type: "GET",
                        url: "https://notebook.genepattern.org/services/sharing/notebooks/",
                        dataType: "json",
                        cache: false,
                        success: function(data) {
                            const notebooks = sort_notebooks(data.results);
                            notebooks.forEach(function(nb) {
                                if (!workshop(nb)) {
                                $(`<div class="col-sm-3" style="padding: 15px; background-color: white; border: 1px solid lightgrey;height: 150px;overflow:hidden;"><h4 class="notebook-name" style="color: #103875; font-size: 17px; font-weight: 600; margin: 0px auto; margin-bottom: 5px;"><a href="https://notebook.genepattern.org/services/sharing/notebooks/${nb.id}/preview/" target="_blank" style="text-decoration:none;">${nb.name}</a></h4>
                                <p class="notebook-description" style="font-size: 12px; margin-bottom: 0px;">${nb.description}</p></div>`)
                                    .appendTo("#notebook_list");
                                }
                            });
                        },
                        error: function(error) {
                            console.log(error);
                        }
                    });
                </script>
            </div> -->
            <br>
        </div>



        <div class="importing-data" id="importing-data">
            <h2>Importing Data</h2>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool Name</th>
                        <th>Tutorials</th>
                        <th>Notebook</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="">TCGA Importer</a></th>
                        <td><a href="https://github.com/genepattern/TCGAImporter/blob/master/how_to_download_a_manifest_and_metadata.pdf">How to download a manifest and metadata <i class="fas fa-file-pdf"></i></a></td>
                        <td><a href="https://notebooks.genepattern.org/services/sharing/notebooks/200/preview/">Bladder Urothelial Carcinoma (BLCA)</a></td>
                        <td>This module imports data from TCGA by taking in a GDC manifest file, downloading the files listed on that manifest, renaming them to be human-friendly, and compiling them into a GCT file to be computer-friendly. <a href="https://www.genepattern.org/modules/docs/TCGAImporter/6">Documentation</a></td>
                    </tr>
                    <tr>
                        <th><a href="">Globus Client</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                    <tr>
                        <th><a href="">Expression Matrix</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                </tbody>
            </table>
        </div>
        <br>
        <div class="gene-expression" id="gene-expression">
            <h2>Gene Expression Analysis</h2>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Module</th>
                        <th>Input</th>
                        <th>Normalization</th>
                        <th>Batch / Confounder Removal</th>
                        <th>Clustering</th>
                        <th>Test Statistic</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="">ComparativeMarkerSelection</a></th>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#GCT">GCT</a> or <a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#RES">RES</a></td>
                        <td></td>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#CLS">CLS</a></td>
                        <td></td>
                        <td>t-test; signal-to-noise; paired t-test</td>
                        <td>Identify differentially expressed genes that can discriminate between distinct classes of samples. <a href="https://www.genepattern.org/modules/docs/ComparativeMarkerSelection/10">Documentation</a>; <a href="https://github.com/genepattern/ComparativeMarkerSelection">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">DESeq2</a></th>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#GCT">GCT</a></td>
                        <td>DESeq2</td>
                        <td></td>
                        <td>Yes</td>
                        <td>Wald test</td>
                        <td>A Differential Expression workflow for RNA-Seq data based on the DESeq2 package from Bioconductor. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Love <i>et al.</i> 2014</a>; <a href="https://genepattern.github.io/DESeq2/v1/index.html">Documentation</a>; <a href="https://github.com/genepattern/DESeq2">Github <i class="fab fa-github"></i></a>
                        </td>
                    </tr>
                    <tr>
                        <th><a href="">Cuffdiff</a></th>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#SAM">SAM</a> or <a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#BAM">BAM</a>; and <a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#GTF">GTF</a></td>
                        <td>geometric; quartile; classic-fpkm</td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td>Finds significant changes in transcript expression, splicing, and promoter use. <a href="http://www.nature.com/nbt/journal/v31/n1/full/nbt.2450.html">Trapnell <i>et al.</i> 2013</a>; <a href="https://www.genepattern.org/modules/docs/Cuffdiff/7">Documentation</a></a>
                        </td>
                    </tr>
                    <tr>
                        <th><a href="">Cufflinks</a></th>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#SAM">SAM</a> or <a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#BAM">BAM</a>; and <a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#GTF">GTF</a></td>
                        <td>geometric; quartile; classic-fpkm</td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td>Assembles transcripts, estimates abundances, and tests for differential expression and regulation in RNA-seq samples. <a href="http://www.nature.com/nbt/journal/v31/n1/full/nbt.2450.html">Trapnell <i>et al.</i> 2013</a>; <a href="https://www.genepattern.org/modules/docs/Cufflinks/7">Documentation</a></a>
                        </td>
                    </tr>
                    <tr>
                        <th><a href="">ExpressionFileCreator</a></th>
                        <td><a href="https://www.genepattern.org/cancer/software/genepattern/file-formats-guide#CEL">CEL</a></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td> Creates a RES or GCT file from a set of Affymetrix CEL files. For IVT arrays only; use AffySTExpressionFileCreator for ST arrays. <a href="https://www.genepattern.org/modules/docs/ExpressionFileCreator/13">Documentation</a></a>
                        </td>
                    </tr>
                    <tr>
                        <th>HierarchicalClustering</th>
                    </tr>
                    <tr>
                        <th>ConsensusClustering</th>
                    </tr>
                    <tr>
                        <th>KMeansClustering</th>
                    </tr>
                    <tr>
                        <th>NMFConsensus</th>
                    </tr>
                    <tr>
                        <th>Bowtie</th>
                    </tr>
                    <tr>
                        <th>FastQC</th>
                    </tr>
                    <tr>
                        <th>Kallisto</th>
                    </tr>
                    <tr>
                        <th>TopHat</th>
                    </tr>
                    <tr>
                        <th>STAR</th>
                    </tr>
                    <tr>
                        <th>DiSCoVER</th>
                    </tr>
                    <tr>
                        <th>Scripture</th>
                    </tr>
                </tbody>
            </table>
        </div>
        <div class="single-cell" id="single-cell">
            <h2>Single-cell Sequencing</h2>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Module</th>
                        <th>Input</th>
                        <th>Normalization</th>
                        <th>Batch / Confounder Removal</th>
                        <th>Clustering</th>
                        <th>Classification / Marker	</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="">SeruatClustering</a></th>
                        <td><a href="">RDS</a></td>
                        <td>log-normal; sctransform </td>
                        <td>SNNAnchor (mutual nearest neighbors)</td>
                        <td>Shared nearest neighbors</td>
                        <td>Yes</td>
                        <td>Companion module for the <a href="https://notebook.genepattern.org/services/sharing/notebooks/376/preview/">Seurat notebook <i class="fas fa-file-alt"></i></a>. Performs clustering and marker identification on single cell data. <a href="https://www.nature.com/articles/nbt.4096">Butler <i>et al.</i> 2018</a>; <a href="https://genepattern.github.io/SeuratClustering/">Documentation</a>; <a href="https://github.com/genepattern/SeuratClustering">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">Conos (Preprocess, Cluster)</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                    <tr>
                        <th><a href="">ScanpyUtilities</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                    <tr>
                        <th><a href="">Amaretto</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                    <tr>
                        <th><a href="">STREAM</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                </tbody>
            </table>
        </div>

        <div class="gsea" id="gsea">
            <h2>Gene Set Enrichment Analysis (GSEA)</h2>
            <ul>
                <li>GSEA (classic</li>
                <li>ssGSEAProjection</li>
                <li>GSEAPreranked</li>
                <li>ConstellationMap</li>
            </ul>
        </div>
        <div class="mutation" id="mutation">
            <h2>Mutational Significane Analysis</h2>
            <ul>
                <li>MutPanning</li>
                <li>MutSigCV</li>
            </ul>
        </div>
        <div class="proteomics" id="proteomics">
            <h2>Proteomics</h2>
            <ul>
                <li>DAPPLE</li>
                <li>ProteomicsAnalysis</li>
                <li>PTM_SEA</li>
                <li>AuDIT</li>
                <li>Peaks</li>
                <li>ProteoArray</li>
            </ul>
        </div>
        <div class="pathway" id="pathway">
            <h2>Pathway Analysis</h2>
            <ul>
                <li>ARACNE</li>
                <li>MINDY</li>
            </ul>
        </div>
        <div class="epigenetics" id="epigenetics">
            <h2>Epigenetics</h2>
            <ul>
                <li></li>
            </ul>
        </div>
        <div class="snp" id="snp">
            <h2>SNP and Copy Number Analysis</h2>
            <ul>
                <li>GISTIC_2.0</li>
                <li>CBS</li>
                <li>HAPSEG</li>
                <li>CNToXCN</li>
                <li>CopyNumberDivideByNormals</li>
                <li>LOHPaired</li>
                <li>SNPFileCreator</li>
                <li>SNPFileSorter</li>
                <li>SNPMultipleSampleAnalysis</li>
                <li>XChromosomeCorrect</li>
            </ul>
        </div>
        <div class="visual" id="visual">
            <h2>Visualization Tools</h2>
            <ul>
                <li>igv.js</li>
                <li>CytoscapeViewer</li>
                <li>HeatMapViewer</li>
                <li>HierarchicalClusteringViewer</li>
            </ul>
        </div>
    </div>

    <h4>Our Team</h4>
            <p></p>

            <table class="table table-striped">
                <tbody>
                    <tr>
                        <td>Jill P. Mesirov - Principal Investigator</td>
                        <td></td>
                    </tr>
                    <tr>
                        <td>Barbara Hill</td>
                        <td>Michael Reich</td>
                    </tr>
                    <tr>
                        <td>Ted Liefeld</td>
                        <td>Thorin Tabor</td>
                    </tr>
                    <tr>
                        <td>Edwin Juarez</td>
                        <td>Helga Thorvaldsdóttir</td>
                    </tr>
                    <tr>
                        <td>Forrest Kim</td>
                        <td></td>
                    </tr>
                </tbody>
            </table>

    <ul class="nav nav-tabs md-tabs" id="myTabMD" role="tablist">
        <li class="nav-item">
          <a class="nav-link active" id="home-tab-md" data-toggle="tab" href="#home-md" role="tab" aria-controls="home-md"
            aria-selected="true">Home</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="profile-tab-md" data-toggle="tab" href="#profile-md" role="tab" aria-controls="profile-md"
            aria-selected="false">Profile</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" id="contact-tab-md" data-toggle="tab" href="#contact-md" role="tab" aria-controls="contact-md"
            aria-selected="false">Contact</a>
        </li>
      </ul>
      <div class="tab-content card pt-5" id="myTabContentMD">
        <div class="tab-pane fade show active" id="home-md" role="tabpanel" aria-labelledby="home-tab-md">
          <p>Raw denim you probably haven't heard of them jean shorts Austin. Nesciunt tofu stumptown aliqua,
            retro synth master cleanse. Mustache cliche tempor, williamsburg carles vegan helvetica.
            Reprehenderit butcher retro keffiyeh dreamcatcher synth. Cosby sweater eu banh mi, qui irure terry
            richardson ex squid. Aliquip placeat salvia cillum iphone. Seitan aliquip quis cardigan american
            apparel, butcher voluptate nisi qui.</p>
        </div>
        <div class="tab-pane fade" id="profile-md" role="tabpanel" aria-labelledby="profile-tab-md">
          <p>Food truck fixie locavore, accusamus mcsweeney's marfa nulla single-origin coffee squid.
            Exercitation +1 labore velit, blog sartorial PBR leggings next level wes anderson artisan four loko
            farm-to-table craft beer twee. Qui photo booth letterpress, commodo enim craft beer mlkshk aliquip
            jean shorts ullamco ad vinyl cillum PBR. Homo nostrud organic, assumenda labore aesthetic magna
            delectus mollit. Keytar helvetica VHS salvia yr, vero magna velit sapiente labore stumptown. Vegan
            fanny pack odio cillum wes anderson 8-bit, sustainable jean shorts beard ut DIY ethical culpa terry
            richardson biodiesel. Art party scenester stumptown, tumblr butcher vero sint qui sapiente accusamus
            tattooed echo park.</p>
        </div>
        <div class="tab-pane fade" id="contact-md" role="tabpanel" aria-labelledby="contact-tab-md">
          <p>Etsy mixtape wayfarers, ethical wes anderson tofu before they sold out mcsweeney's organic lomo
            retro fanny pack lo-fi farm-to-table readymade. Messenger bag gentrify pitchfork tattooed craft beer,
            iphone skateboard locavore carles etsy salvia banksy hoodie helvetica. DIY synth PBR banksy irony.
            Leggings gentrify squid 8-bit cred pitchfork. Williamsburg banh mi whatever gluten-free, carles
            pitchfork biodiesel fixie etsy retro mlkshk vice blog. Scenester cred you probably haven't heard of
            them, vinyl craft beer blog stumptown. Pitchfork sustainable tofu synth chambray yr.</p>
        </div>
      </div>


</body>
</html>