<html lang="en">
<head>
    <title>Single Cell GenePattern Resources</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" target="_blank" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
    <link rel="stylesheet" target="_blank" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">

    <!-- Style -->
    <link rel="stylesheet" href="css/style.css">
</head>
<body>
    <div class="main-container">
        <div class="header">
            <h1>Single Cell GenePattern Resources</h1>
            <hr>
        </div>
        <div class="toc">
            <h4>Table of Contents</h4>
            <ol>
                <li><a href="#get-started">Get Started</a></li>
                <li><a href="#preprocessing">Preprocessing</a></li>
                <ol>
                <li><a href="#alignment">Alignment</a></li>
                <li><a href="#qc">QC/Filtering</a></li>
                </ol>
                <li><a href="#single-cell">Single Cell Analysis</a></li>
                <ol>
                    <li><a href="#gsea">Seurat Suite</a></li>
                    <li><a href="#mutation">STREAM Suite</a></li>
                    <li><a href="#mutation">CONOS Suite</a></li>
                    <li><a href="#mutation">Monocle</a></li>
                    <li><a href="#mutation">ScanpyUtilities</a></li>
                </ol>
                <li><a href="#visual">Visualization Tools</a></li>
                <li><a href="">Contributors</a></li>
                
            </ol>
        </div>
        <div class="get-started" id="get-started">
            <h2>Get Started</h2>
            <p>If you are new to GenePattern Notebook, we recommend checking out some of our tutorial notebooks below:</p>
        </div>
        
        <div class="importing-data" id="importing-data">
            <h3>Tutorial Notebooks</h3>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Notebook</th>
                        <th>Description</th>
                        <th>Link</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/361/preview/">GenePattern Notebook Tutorial</a></th>
                        <td>A tutorial on how to use the basic features of the GenePattern Notebook environment.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/361/preview/"><i class="fas fa-file-alt"></i></a></td>
                    </tr>
                    <tr>
                        <th><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/363/preview/">Introduction to Text and Document</a></th>
                        <td>A tutorial on the text/document features specific to GenePattern Notebook</td>
                        <td style="text-align: center;"><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/363/preview/"><i class="fas fa-file-alt"></i></a></td>                    
                    </tr>  
                    <tr>
                        <th><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/294/preview/">UI Builder Tutorial</a></th>
                        <td>A tutorial of how to use the UI Builder, aimed at notebook authors with programmatic skill.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/294/preview/"><i class="fas fa-file-alt"></i></a></td>                    
                    </tr>   
                    <tr>
                        <th><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/362/preview/">Programmatic Features</a></th>
                        <td>A tutorial on the notebook features catering to programmers</td>
                        <td style="text-align: center;"><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/362/preview/"><i class="fas fa-file-alt"></i></a></td>
                    </tr> 
                    <tr>
                        <th><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/29/preview/">GenePattern Files in Python</a></th>
                        <td>A tutorial on how to programmatically use GenePattern files in conjunction with common Python libraries, such as csv, numpy, pandas and matplotlib.</td>
                        <td style="text-align: center;"><a href=""><i class="fas fa-file-alt"></i></a></td>                    
                    </tr>    
                    <!-- <tr>
                        <th><a href=""></a></th>
                        <td></td>
                        <td style="text-align: center;"><a href=""><i class="fas fa-file-alt"></i></a></td>                    
                    </tr>             -->
                </tbody>
            </table>
        </div>
        <hr>
        
        <h2 class="preprocessing" id="preprocessing">Preprocessing</h2>
        <div class="alignment" id="alignment">
            <h3>Alignment</h3>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool</th>
                        <th>Input</th>
                        <th>Output</th>
                        <th>Quantitation</th>
                        <th>Details</th>
                        <th>Documentation</th>
                        <th>Reference</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="https://cloud.genepattern.org/gp/module/doc/urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:999999999">STAR.Aligner</a></th>
                        <td>FASTA/FASTQ</td>
                        <td>SAM/BAM</td>
                        <td></td>
                        <td>STAR (Spliced Transcripts Alignment to a Reference) 2.5.3a is a fast RNA-seq reads to genome mapper.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://cloud.genepattern.org/gp/module/doc/urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00370:999999999"><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href="http://www.ncbi.nlm.nih.gov/pubmed/23104886"><i class="fas fa-newspaper"></i></a></td>
                    </tr>
                    <tr>
                        <th><a>Kallisto</a></th>
                        <td>FASTA/FASTQ</td>
                        <td>TSV/CSV (Transcript-level and gene-level abundance)</td>
                        <td style="text-align: center;"><i class="fa fa-check"></i></td>
                        <td>Bare-bones implementation of kallisto, intended to be used for RNA quantitation against a human index.</a> 
                        </td>
                        <td style="text-align: center;"><a target="_blank" href="https://github.com/genepattern/Kallisto"><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>
                    <!-- <tr>
                        <th><a>Salmon</a></th>
                        <td>FASTA/FASTQ</td>
                        <td>TSV/CSV (Transcript-level and gene-level abundance)</td>
                        <td style="text-align: center;"><i class="fa fa-check"></i></td>
                        <td></td>
                    </tr> -->
                    <tr>
                        <th><a>HISAT2Aligner</a></th>
                        <td>FASTA/FASTQ</td>
                        <td>SAM</td>
                        <td></td>
                        <td>HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome).</td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>
                    <tr>
                        <th><a>TopHat</a></th>
                        <td>FASTA/FASTQ</td>
                        <td>BAM/BED</td>
                        <td></td>
                        <td>TopHat 2.0.11 is a fast splice junction mapper for RNA-Seq reads.</td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>
                </tbody>
            </table>
            <div class="alert alert-danger">left out salmon bc it's not ready</div>
        </div>
        <div class="qc" id="qc">
            <h3>QC/Filtering</h3>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool</th>
                        <th>Input</th>
                        <th>Filtering</th>
                        <th>Normalization</th>
                        <th>Details</th>
                        <th>Documentation</th>
                        <th>Reference</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a>FastQC</a></th>
                        <td>FASTQ, SAM, or BAM</td>
                        <td></td>
                        <td></td>
                        <td>FastQC takes sequencing data as input and returns a report on read quality.</td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>    
                    <tr>
                        <th><a>Seurat.QC</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td>Seurat.QC allows you to explore QC metrics based on any user-defined criteria. A few QC metrics commonly used by the community include: unique feature counts, mitochondrial counts, total number of molecules detected within a cell. <a target="_blank" href="https://satijalab.org/seurat/v3.2/pbmc3k_tutorial.html">Source</a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr> 
                    <tr>
                        <th><a>Seurat.Preprocessing</a></th>
                        <td></td>
                        <td><i class="fa fa-check"></i></td>
                        <td>log-normalize</td>
                        <td>Seurat.Preprocessing allows you to explore QC metrics and filter cells based on any user-defined criteria. See Seurat.QC for more info.</td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr> 
                    <tr>
                        <th><a>Trimmomatic</a></th>
                        <td></td>
                        <td><i class="fa fa-check"></i></td>
                        <td></td>
                        <td>Provides a variety of options for trimming Illumina FASTQ files of adapter sequences and low-quality reads. </td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr> 
                    <tr>
                        <th><a>ScanpyUtilities</a></th>
                        <td></td>
                        <td><i class="fa fa-check"></i></td>
                        <td><a target="_blank" href="https://genepattern.github.io/ScanpyUtilities/">See Documentation</a></td>
                        <td>Single-cell preprocessing (annotation and filtering) GenePattern module using Scanpy</td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>  
                    <tr>
                        <th><a>HTSeq.Count</a></th>
                        <td>SAM/BAM and GTF/GFF</td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <<td style="text-align: center;"><a target="_blank" href=""><i class="fa fa-book"></i></a></td>
                        <td style="text-align: center;"><a target="_blank" href=""><i class="fas fa-newspaper"></i></a></td>
                    </tr>      
                </tbody>
            </table>
        </div>
        <hr>
        <div class="single-cell" id="single-cell">
            <h2>Single Cell Analysis</h2>
            <h3>Seurat Suite</h3>
            <p>Seurat (<a target="_blank" href="https://satijalab.org/seurat/">https://satijalab.org/seurat/</a>) is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.
            <br><br>
            Within GenePattern, Seurat is accessed as several different modules exposing different portions of the Seurat workflow; <b>Seurat.Preprocessing</b>, <b>Seurat.BatchCorrection</b>, <b>Seurat.QC</b> and <b>Seurat.Clustering</b>.</p>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool</th>
                        <th>Description</th>
                        <th>Documentation</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a target="_blank" href="https://github.com/genepattern/Seurat.QC/blob/develop/doc.md">Seurat.QC<a></th>
                        <td>The <b>Seurat.QC</b> Module aims to provide a quick look at some basic single-cell datasets. This is typically the first step in the Seurat Suite modules. The resultant preprocessed dataset can then be used for downstream analyses and visualizations (such as <b>Seurat.Preprocessing</b>). If you are doing batch correction, then you want to run <b>Seurat.BatchCorrection</b> before <b>Seurat.QC</b>.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://github.com/genepattern/Seurat.QC/blob/develop/doc.md"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a target="_blank" href="https://github.com/genepattern/Seurat.Preprocessing/blob/develop/doc.md">Seurat.Preprocessing</a></th>
                        <td>The <b>Seurat.Preprocessing</b> Module aims to provide a way to integrate the multiple steps in the preprocessing of single-cell datasets. The resultant preprocessed dataset can then be used for downstream analyses and visualizations (<b>Seurat.Clustering</b>). You may need to run this module multiple times if you want to change the filtering step.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://github.com/genepattern/Seurat.Preprocessing/blob/develop/doc.md"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a target="_blank" href="https://genepattern.github.io/SeuratClustering/">SeuratClustering</a></th>
                        <td><b>Seurat.Clustering</b> performs UMAP clustering and marker identification on single-cell RNA-Seq data.
                        <br><br>
                        The <b>Seurat.Clustering</b> module performs uses Seurat version 3.0.2. It performs following steps from Seurat to the input dataset:
                        <ol>
                            <li>FindNeighbors</li>
                            <li>FindClusters</li>
                            <li>RunUMAP</li>
                            <li>DimPlot</li>
                        </ol>
                            For details of the Seurat implementation please refer to the <a target="_blank" href="https://satijalab.org/seurat/">Seurat documentation</a> from the Satija lab. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/SeuratClustering/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a target="_blank" href="https://github.com/genepattern/Seurat.BatchCorrection/blob/develop/doc.md">Seurat.BatchCorrection</a></th>
                        <td>The <b>Seurat.BatchCorrection</b> Module aims to integrate / "batch-correct" multiple single-cell datasets and identify shared cell states that are present across different datasets, regardless of their origin. Once the Module integrates these datasets, the returned object will contain a new Assay that holds an integrated / batch-corrected expression matrix for all cells. The resultant batch-corrected expression matrix can then be used for downstream analyses and visualizations.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://github.com/genepattern/Seurat.BatchCorrection/blob/develop/doc.md"><i class="fa fa-book"></i></a></td>
                    </tr>
                </tbody>
            </table>

            <h3>STREAM Suite</h3>
            <p>STREAM (Single-cell Trajectories Reconstruction, Exploration And Mapping) (<a target="_blank" href="https://github.com/pinellolab/STREAM">https://github.com/pinellolab/STREAM</a>) is an interactive pipeline capable of disentangling and visualizing complex branching trajectories from both single-cell transcriptomic and epigenomic data. Within GenePattern STREAM is implemented as a collection of modules that cover the entire STREAM processing pipeline to allow individual steps to be performed interactively for data exploration.</p>
            <div style="text-align: center;" class="row">
                <a target="_blank" href="https://notebooks.genepattern.org/services/sharing/notebooks/355/preview/"><img src="stream-ss2.png" width="50%"></a>
                <button class="btn" src="">STREAM Notebook</button>
            </div>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool</th>
                        <th>Description</th>
                        <th>Documentation</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.Preprocess/">STREAM.Preprocess</a></th>
                        <td><b>STREAM.Preprocess</b> will prepare single cell RNA-seq data for the STREAM trajectory analysis. It will normalize, log transform, filter and remove mitochondiral genes.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.Preprocess/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.FeatureSelection/">STREAM.FeatureSelection</a></th>
                        <td><b>STREAM.FeatureSelection</b> is used to identify features to be used in the downstream analysis. Two types of features can be used;
                        <ol>
                            <li>Variable genes</li>
                            <li>Top principal components</li>
                        </ol>
                        </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.FeatureSelection/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.DimensionReduction/">STREAM.DimensionReduction</a></th>
                        <td><b>STREAM.DimensionReduction</b> is used to reduce the dimensionality of the dataset to be used in the downstream analysis.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.DimensionReduction/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.Plot2DVisualization/">STREAM.Plot2DVisualization</a></th>
                        <td><b>STREAM.Plolt2DVisualization</b> is used check if there is clear meaningful trajectory pattern to the data. If there is, we will continue the downstream analysis placing the cells onto the trajectories. If not, we would go back to previous steps to modify the parameters used to filter and prepare the data to try different settings.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.Plot2DVisualization/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.SeedEPGStructure/">STREAM.SeedEPGStructure</a></th>
                        <td><b>STREAM.SeedEPGStructure</b> is use to seed the initial elastic principal graph prior to starting the trajectory learning process. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.SeedEPGStructure/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.ElasticPrincipalGraph/">STREAM.ElasticPrincipalGraph</a></th>
                        <td><b>STREAM.ElasticPrincipalGraph</b> uses elastic principal graph learning to calculate a pseudotime trajectory. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.ElasticPrincipalGraph/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.EPGAdjustFinalGraph/">STREAM.EPGAdjustFinalGraph</a></th>
                        <td><b>STREAM.EPGAdjustFinalGraph</b> facilitates some final adjustments to the graph generated by <b>STREAM.ElasticPrincipalGraph</b> such as optimizing the structure, pruning the final structure or extending leaf nodes. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.EPGAdjustFinalGraph/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.VisualizeTrajectories/">STREAM.VisualizeTrajectories</a></th>
                        <td><b>STREAM.VisualizeTrajectories</b> is used to generate the subway and stream plots for a computed graph structure. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.VisualizeTrajectories/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.DetectDifferentiallyExpressedGenes/">STREAM.DetectDifferentiallyExpressedGenes</a></th>
                        <td><b>STREAM.DetectDifferentiallyExpressedGenes</b> used to detect differentially expressed genes between pairs of branches.</td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.DetectDifferentiallyExpressedGenes/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.DetectLeafGenes/">STREAM.DetectLeafGenes</a></th>
                        <td><b>STREAM.DetectLeafGenes</b> is used to detect marker genes for each leaf branch. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.DetectLeafGenes/"><i class="fa fa-book"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="https://genepattern.github.io/STREAM.DetectTransitionGenes/">STREAM.DetectTransitionGenes</a></th>
                        <td><b>STREAM.DetectTransitionGenes</b> is used to detect marker genes for each transition. </td>
                        <td style="text-align: center;"><a target="_blank" href="https://genepattern.github.io/STREAM.DetectTransitionGenes/"><i class="fa fa-book"></i></a></td>
                    </tr>
                </tbody>
            </table>
            <div class="alert alert-danger">left out monocle for now</div>
            <!-- <h3>Monocle</h3>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Tool</th>
                        <th>Description</th>
                        <th>Documentation</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th></th>
                        <td></td>
                        <td></td>
                    </tr>
                </tbody>
            </table> -->

            <h3>ScanpyUtilities</h3>
            <p>
                <a target="_blank" href="https://scanpy.readthedocs.io/en/stable/">Scanpy</a> is a scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.
                <br><br>
                Some common single-cell preprocessing steps require the entire dataset to be loaded into memory which can be an issue on less powerful machines or with larger datasets. This makes it useful to expose ScanPy as a genepattern module where the execution can be offloaded to different, larger, compute resources to perform initial steps to reduce the dataset size before extensive, interactive visualization such as might be done in a GenePattern notebook.
                <br><br>
                <a target="_blank" href="https://genepattern.github.io/ScanpyUtilities/">Documentation <i class="fa fa-book"></i></a>
                <br>
                <a target="_blank" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0">Wolf <i>et al.</i> 2018</a>
                <br><br>
                This module exposes many functions from scanpy version 1.3.3 to be used as a GenePattern module:
            </p>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Function</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th>Annotation</th>
                        <td>Add count information to the data file</td>
                    </tr>
                    <tr>
                        <th>Cell Filtering</th>
                        <td>Filter out cells based on the various thresholds.
                        </td>
                    </tr>
                    <tr>
                        <th>Gene Filtering</th>
                        <td>Filter out genes based on the various thresholds.</td>
                    </tr>
                    <tr>
                        <th>Cell Type Identification</th>
                        <td>Run the cell type identification pipeline provided in the R package garnett</td>
                    </tr>
                    <tr>
                        <th>Normalization</th>
                        <td>Run the single-cell normalization pipeline provided in the R package scran</td>
                    </tr>
                    <tr>
                        <th>High Variance Genes</th>
                        <td>Filter and subset the data to retain only the N most variable genes.</td>
                    </tr>
                    <tr>
                        <th>Dimension Reduction</th>
                        <td>Compute UMAP or TSNE coordinates for visualization.</td>
                    </tr>
                </tbody>
            </table>

        </div>
        <hr>
        <!-- <div class="single-cell" id="single-cell">
            <h2>Single Cell Analysis</h2>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Module</th>
                        <th>Input</th>
                        <th>Normalization</th>
                        <th>Batch / Confounder Removal</th>
                        <th>Clustering</th>
                        <th>Classification / Marker	</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="">SeuratClustering</a></th>
                        <td><a href="">RDS</a></td>
                        <td>log-normal; sctransform </td>
                        <td>SNNAnchor (mutual nearest neighbors)</td>
                        <td>Shared nearest neighbors</td>
                        <td>Yes</td>
                        <td><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/376/preview/">Seurat notebook <i class="fas fa-file-alt"></i></a>. Performs clustering and marker identification on single cell data. <a target="_blank" href="https://www.nature.com/articles/nbt.4096">Butler <i>et al.</i> 2018</a>; <a target="_blank" href="https://genepattern.github.io/SeuratClustering/">Documentation</a>; <a target="_blank" href="https://github.com/genepattern/SeuratClustering">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">CONOS (Preprocess, Cluster)</a></th>
                        <td>text files with tab-separated gene expression matrices (rows are genes, columns are unique cells)</td>
                        <td>variance normalization</td>
                        <td>Yes</td>
                        <td>Pagoda 2; Leiden community clustering; Walktrap clustering</td>
                        <td>No</td>
                        <td><a target="_blank" href="">CONOS notebook <i class="fas fa-file-alt"></i></a> . Clustering on Networks of Samples for Single Cell RNA-sequencing <a target="_blank" href="https://doi.org/10.1038/s41592-019-0466-z">Barkas <i>et al.</i> 2019</a>; <a target="_blank" href="">Documentation</a>; <a target="_blank" href="https://github.com/kharchenkolab/conos">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">ScanpyUtilities</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                    <tr>
                        <th><a href="">STREAM</a></th>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                        <td></td>
                    </tr>
                </tbody>
            </table>
        </div>

        <div class="gsea" id="gsea">
            <h2>Gene Set Enrichment Analysis (GSEA)</h2>
            <table class="table table-striped">
                <thead>
                    <tr>
                        <th>Module</th>
                        <th>Input</th>
                        <th>Sample Normalization</th>
                        <th>Reduces Dimensionality</th>
                        <th>RNA-seq?</th>
                        <th>Description</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <th><a href="">GSEA (classic)</a></th>
                        <td>GCT or RES</td>
                        <td>meandiv; none</td>
                        <td></td>
                        <td>not recommended</td>
                        <td><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/370/preview/">GSEA Notebook <i class="fas fa-file-alt"></i></a> Gene Set Enrichment Analysis for the GenePattern platform. <a target="_blank" href="https://gsea-msigdb.github.io/gsea-gpmodule/v20/index.html">Documentation</a>; <a target="_blank" href="">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">ssGSEA</a></th>
                        <td>GCT (minimum of two samples)</td>
                        <td>none; rank; log rank; log</td>
                        <td><i class="fa fa-check"></i></td>
                        <td><i class="fa fa-check"></i></td>
                        <td><a target="_blank" href="https://notebook.genepattern.org/services/sharing/notebooks/386/preview/">ssGSEA Notebook <i class="fas fa-file-alt"></i></a>. Performs single sample GSEA on the GenePattern platform. <a target="_blank" href="https://gsea-msigdb.github.io/ssGSEA-gpmodule/v10/index.html">Documentation</a>; <a target="_blank" href="">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">GSEAPreranked</a></th>
                        <td><a target="_blank" href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#RNK:_Ranked_list_file_format_.28.2A.rnk.29">RNK</a> = ranked list of genes</td>
                        <td>meandiv; none</td>
                        <td></td>
                        <td><i class="fa fa-check"></i></td>
                        <td>Runs the gene set enrichment analysis against a user-supplied ranked list of genes. <a target="_blank" href="https://gsea-msigdb.github.io/gseapreranked-gpmodule/v7/index.html">Documentation</a>; <a target="_blank" href="">Github <i class="fab fa-github"></i></a></td>
                    </tr>
                    <tr>
                        <th><a href="">ConstellationMap</a></th>
                    </tr>
                </tbody>
            </table>
        </div>
        <div class="visual" id="visual">
            <h2>Visualization Tools</h2>
            <ul>
                <li>igv.js</li>
                <li>CytoscapeViewer</li>
                <li>HeatMapViewer</li>
                <li>HierarchicalClusteringViewer</li>
            </ul>
        </div>
    -->
    <h4>Contributions</h4>
        <p></p>

        <table class="table table-striped">
            <tbody>
                <tr>
                    <td>Jill P. Mesirov - Principal Investigator</td>
                    <td></td>
                </tr>
                <tr>
                    <td>Barbara Hill</td>
                    <td>Michael Reich</td>
                </tr>
                <tr>
                    <td>Ted Liefeld</td>
                    <td>Thorin Tabor</td>
                </tr>
                <tr>
                    <td>Edwin Juarez</td>
                    <td>Helga Thorvaldsd√≥ttir</td>
                </tr>
                <tr>
                    <td>Forrest Kim</td>
                    <td></td>
                </tr>
            </tbody>
        </table>
    </div>

</body>
</html>